### PR TITLE
New year's cleanin'

### PR DESCRIPTION
**Pull request by local-minimum**
Key features:

* external js now resides in own folder and has own endpoint
* all endpoints that deliver pages are now in `ui_pages.py`
* in `ui_server.py` now only remains resource-endpoints.
* no more mixed api/page serving endpoints.
* cleaning up the experiment api
* system tests that checks that pages and resources can be accessed.
* fixed failing tests due to new api-endpoints
* removed some unused stuff
* fixed some lint errors.

**Note:** While this does not really introduce any new lint errors, there are a million long lines that appear as changed just because code moved around or lost a bit of indentation. These will not be fixed in this PR.

### CODE REVIEW

File: scanomatic/ui_server_data/qc_norm.html [hunk 1]
```diff
@@ -3,18 +3,19 @@
     <head>
         <title>Quality Control Prototype</title>
         <meta charset="utf-8" />
-        <script src="js/jquery-2.2.3.js"></script>
-        <script src="js/bootstrap.js"></script>
-        <script src="js/d3.js"></script>
+        <script src="js/external/jquery.js"></script>
+        <script src="js/external/bootstrap.js"></script>
+        <script src="js/external/d3.js"></script>
```
Comment (skymandr) [inline scanomatic/ui_server_data/qc_norm.html]: Where do we use d3? I thought you opted for another library? Or does that one depend on d3? General question: are we using minimised versions of externals when we can?

File: scanomatic/ui_server_data/maintain.html [hunk 2]
```diff
@@ -20,61 +19,9 @@ <h2 id="version-info">Checking version...</h2>
             <p><em>
                 If Scan-o-Matic currently have jobs it is working on or in
                 queue, it is generally not recommended to do anything to the
-                server.  In most cases it still should be alright, but some
-                cases remain to be fully tested and verified (especially
-                rebooting server while scanning projects are running).
-                Shutting down the server while scanning experiments are running
-                will always cause issues as the server is needed to acquire
-                scanners.
+                server. You will terminate all.
```
Comment (skymandr) [inline scanomatic/ui_server_data/maintain.html]: all what?

File: scanomatic/ui_server_data/qc_norm.html (hunk 1 above)
Comment (local-minimum) [reply scanomatic/ui_server_data/qc_norm.html]: I think qc uses it still. Or somewhere. It was used anyhow

File: scanomatic/ui_server_data/js/analysis.js [hunk 3]
```diff
@@ -136,24 +136,24 @@ function get_regridding_setting(i) {
...
     path = get_dir();
     get_path_suggestions(
         input,
         true,
-        "",
+        '',
         null,
         function(data, status) {
 
             //TODO: For some reason popup don't appear...
```
Comment (skymandr) [inline scanomatic/ui_server_data/js/analysis.js]: Still an issue?

File: scanomatic/ui_server/management_api.py [hunk 4]
```diff
@@ -1,187 +1,65 @@
...
-                os.kill(proc_id, signal.SIGKILL)
-                server_ids.add(proc_id)
-
-            if rpc_client.online:
-                return jsonify(success=False,
-                               reason="Tried to kill processes {0}, but somehow server is online".format(server_ids))
-
-            return jsonify(success=True)
-
     @app.route("/api/app/<action>", methods=['post', 'get'])
```
Comment (skymandr) [inline scanomatic/ui_server/management_api.py]: This endpoint seems completely superfluous now. Why not just `/api/app/version`?

File: scanomatic/ui_server_data/js/analysis.js (hunk 3 above)
Comment (local-minimum) [reply scanomatic/ui_server_data/js/analysis.js]: Sometimes you have to type stuff for it to start to appear, no real point in working on it, rather fix when upping to react or so.

File: scanomatic/__init__.py [hunk 5]
```diff
@@ -1,19 +1,18 @@
 #!/usr/bin/env python
 """Part of analysis work-flow that holds a grid arrays"""
+import os
 
 __author__ = "Martin Zackrisson"
 __copyright__ = "Swedish copyright laws apply"
 __credits__ = ["Martin Zackrisson", "Mats Kvarnstroem", "Andreas Skyman", ""]
```
Comment (skymandr) [inline scanomatic/__init__.py]: And Gregoire and Fredrik?

File: scanomatic/__init__.py [hunk 6]
```diff
@@ -1,19 +1,18 @@
...
 __license__ = "GPL v3.0"
-__version__ = "v2.0.4"
+__version__ = "v2.1.6"
 __maintainer__ = "Martin Zackrisson"
 __email__ = "martin.zackrisson@gu.se"
```
Comment (skymandr) [inline scanomatic/__init__.py]: No longer valid?

File: scanomatic/ui_server_data/maintain.html (hunk 2 above)
Comment (local-minimum) [reply scanomatic/ui_server_data/maintain.html]: The world :)  Fixed...

File: scanomatic/ui_server_data/qc_norm.html (hunk 1 above)
Comment (skymandr) [reply scanomatic/ui_server_data/qc_norm.html]: ok

File: scanomatic/ui_server/management_api.py (hunk 4 above)
Comment (local-minimum) [reply scanomatic/ui_server/management_api.py]: agree, fixed.

File: scanomatic/__init__.py (hunk 5 above)
Comment (local-minimum) [reply scanomatic/__init__.py]: True that. Perhaps @gdetrez and @dindinoh could PR on it if they want in on the credz.

File: scanomatic/ui_server_data/js/analysis.js (hunk 3 above)
Comment (skymandr) [reply scanomatic/ui_server_data/js/analysis.js]: I agree. Just wondered if perhaps it had been solved as an unforseen side effect! =)

I don't suggest you do it, but do you think an xfailing test could be written for this?

File: scanomatic/__init__.py (hunk 6 above)
Comment (local-minimum) [reply scanomatic/__init__.py]: Version changes aren't really valid but they show in the app and ensure you don't load outdated js/css/fonts/html so I don't think it hurts upping numbers and keeping them.

File: scanomatic/ui_server_data/js/analysis.js (hunk 3 above)
Comment (local-minimum) [reply scanomatic/ui_server_data/js/analysis.js]: It's at system-test level basically... not sure it is worth time it takes to write it.

File: scanomatic/__init__.py (hunk 6 above)
Comment (skymandr) [reply scanomatic/__init__.py]: I meant your email.

File: scanomatic/ui_server_data/js/analysis.js (hunk 3 above)
Comment (skymandr) [reply scanomatic/ui_server_data/js/analysis.js]: Ok. Perhaps make an issue for it if there isn't one?

Comment (local-minimum) [reply scanomatic/ui_server_data/js/analysis.js]: I'll make a task for what would invalidate the issue instead.

File: scanomatic/__init__.py (hunk 6 above)
Comment (local-minimum) [reply scanomatic/__init__.py]: true, don't want mails though :grinning:  
will remove in next PR instead

**Comment by local-minimum** [comment]
Should add I've clicked about and tried doing a lot of the stuff in the UI.

File: scanomatic/__init__.py (hunk 5 above)
Comment (skymandr) [reply scanomatic/__init__.py]: Ok.

File: scanomatic/__init__.py (hunk 6 above)
Comment (skymandr) [reply scanomatic/__init__.py]: Ok.

File: scanomatic/ui_server/management_api.py (hunk 4 above)
Comment (skymandr) [reply scanomatic/ui_server/management_api.py]: Looks good to me now.

File: scanomatic/ui_server_data/js/analysis.js (hunk 3 above)
Comment (skymandr) [reply scanomatic/ui_server_data/js/analysis.js]: Good.

**Pull request merged**